### PR TITLE
fix(jellyfish-api-core, jellyfish-transaction-builder): Fix failed tests for fortcanningcrunch

### PR DESCRIPTION
**Pull request by surangap**
#### Which issue(s) does this PR fixes?:

Fixes #https://github.com/DeFiCh/ain/pull/1243

### CODE REVIEW

File: packages/testcontainers/src/containers/RegTestContainer/index.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ export class RegTestContainer extends DeFiDContainer {
       '-fortcanningheight=8',
       '-fortcanningmuseumheight=9',
       '-fortcanninghillheight=10',
-      '-fortcanningroadheight=11'
+      '-fortcanningroadheight=11',
+      '-fortcanningspicegardenheight=12'
```
Comment (surangap) [inline packages/testcontainers/src/containers/RegTestContainer/index.ts:39]:
```suggestion
      '-fortcanningcrunchheight=12'
```

File: packages/jellyfish-api-core/__tests__/category/account/getAccountHistory.test.ts [hunk 2]
```diff
@@ -44,8 +44,6 @@ describe('Account', () => {
     expect(history.txn).toStrictEqual(referenceHistory.txn)
     expect(history.txid).toStrictEqual(referenceHistory.txid)
     expect(history.type).toStrictEqual(referenceHistory.type)
-    expect(history.blockHash).toStrictEqual(referenceHistory.blockHash)
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/__tests__/category/account/getAccountHistory.test.ts:47]: Might be better to keep another history which in a minted block and non undefined block hash

File: packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts [hunk 3]
```diff
@@ -888,327 +888,328 @@ describe('paybackLoan failed', () => {
...
-      vaultId: vaultId,
-      amounts: '2@TSLA',
-      from: colAddr
-    })
-
-    await expect(promise).resolves.not.toThrow()
-  })
-})
+// Jimmy: Temporarily comment this code, as it should throw error
+// describe('paybackLoan before FortCanningHeight', () => {
```
Comment (ivan-zynesis) [inline packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts:892]: Shall we just `skip` it? Thus remained visible in test report.

File: apps/playground-api/src/setups/setup.gov.ts [hunk 4]
```diff
@@ -64,12 +64,7 @@ export class SetupGov extends PlaygroundSetup<Record<string, any>> {
   }
 
   async has (each: any): Promise<boolean> {
-    const key = Object.keys(each)[0]
-    const gov = await this.client.masternode.getGov(key)
-    if (Object.keys(gov[key]).length > 0) {
-      return true
-    }
-    return false
+    return false // for just overwrite
```
Comment (jingyi2811) [inline apps/playground-api/src/setups/setup.gov.ts:67]: If return `false` is something expected, we should remove this function

Comment (jingyi2811) [reply apps/playground-api/src/setups/setup.gov.ts:67]: And the function name is `has`,  it is a positive word so return true seems more sensible

File: apps/playground-api/src/setups/setup.ts [hunk 5]
```diff
@@ -33,10 +33,6 @@ export abstract class PlaygroundSetup<Each> {
     await this.before(list)
 
     for (const each of list) {
-      if (await this.has(each)) {
```
Comment (jingyi2811) [inline apps/playground-api/src/setups/setup.ts:36]: @canonbrother Here, I changed your code.  As this has function always returns false, this code can be removed.

File: packages/jellyfish-api-core/__tests__/category/vault/placeAuctionBid.test.ts [hunk 6]
```diff
@@ -464,13 +478,13 @@ describe('placeAuctionBid success', () => {
           batches: [
             {
               index: 0,
-              collaterals: ['5004.44437154@DFI', '0.49955556@BTC'], // 5004.44437154 + 4995.5556 = 9999.99997154
-              loan: '499.56012216@TSLA' // 499.56012216 * 15 = 7493.4018324
+              collaterals: ['5004.44449284@DFI', '0.49955555@BTC'],
```
Comment (jingyi2811) [inline packages/jellyfish-api-core/__tests__/category/vault/placeAuctionBid.test.ts:481]: @surangap  All these values should remain the same.

File: apps/playground-api/src/setups/setup.ts (hunk 5 above)
Comment (canonbrother) [reply apps/playground-api/src/setups/setup.ts:36]: don't change that code.. its by designed

**Review by canonbrother — CHANGES_REQUESTED** [review]
https://github.com/JellyfishSDK/jellyfish/pull/1461/files#r882297326

File: apps/playground-api/src/setups/setup.ts (hunk 5 above)
Comment (jingyi2811) [reply apps/playground-api/src/setups/setup.ts:36]: Okay

File: packages/jellyfish-api-core/__tests__/category/vault/placeAuctionBid.test.ts (hunk 6 above)
Comment (surangap) [reply packages/jellyfish-api-core/__tests__/category/vault/placeAuctionBid.test.ts]: Thought this was reverted back. should be fine now.

**Review by jingyi2811 — APPROVED** [review]

**Review by canonbrother — DISMISSED** [review]

**Review by fuxingloh — APPROVED** [review]

**Comment by fuxingloh** [comment]
I will update `activateAfterBlock` on WhaleApi controller such that it returns `activateAfterBlock: 0` to prevent downstream regression.

**Pull request merged**